draft-ietf-tls-external-psk-importer.md

```diff
@@ -79,158 +80,159 @@ when, and only when, they appear in all capitals, as shown here.
-for each hash function used by each ciphersuite they support. For example, importing a key for
-TLS_AES_128_GCM_SHA256 and TLS_AES_256_GCM_SHA384 would yield two PSKs, one for SHA256 and another
-for SHA384. In contrast, if TLS_AES_128_GCM_SHA256 and TLS_CHACHA20_POLY1305_SHA256 are supported,
-only one derived key is necessary.
+Endpoints generate a compatible ipskx for each target ciphersuite they offer. For example, importing a
+key for TLS_AES_128_GCM_SHA256 and TLS_AES_256_GCM_SHA384 would yield two PSKs, one for HKDF-SHA256 and
+another for HKDF-SHA384. In contrast, if TLS_AES_128_GCM_SHA256 and TLS_CHACHA20_POLY1305_SHA256
+are supported, only one derived key is necessary.
 
 The resulting IPSK base key 'ipskx' is then used as the binder key in TLS 1.3 with identity
-ImportedIdentity. With knowledge of the supported hash functions, one may import PSKs before
-the start of a connection.
+ImportedIdentity. With knowledge of the supported KDFs, one may import PSKs before the start
+of a connection.
 
 EPSKs may be imported for early data use if they are bound to protocol settings and configurations that would
 otherwise be required for early data with normal (ticket-based PSK) resumption. Minimally, that means ALPN,
 QUIC transport settings, etc., must be provisioned alongside these EPSKs.
 
-# Label Values
-
-For clarity, the following table specifies PSK importer labels for varying instances of the TLS handshake.
-
-| Protocol   | Label   |
-|:----------:|:-------:|
-| TLS 1.3 {{RFC8446}} | "tls13" |
-| QUICv1 {{I-D.ietf-quic-transport}} | "tls13" |
-| TLS 1.2 {{RFC5246}} | "tls12" |
-| DTLS 1.2 {{!RFC6347}} | "dtls12" |
-| DTLS 1.3 {{!I-D.ietf-tls-dtls13}} | "dtls13" |
-
 # Deprecating Hash Functions
 
-If a client or server wish to deprecate a hash function and no longer use it for TLS 1.3, they may remove this
-hash function from the set of hashes used during while importing keys. This does not affect the KDF operation
-used to derive concrete PSKs.
+If a client or server wish to deprecate a hash function and no longer use it for TLS 1.3,
+they remove the corresponding KDF from the set of target KDFs used for importing keys.
+This does not affect the KDF operation used to derive Imported PSKs.
 
-# Backwards Compatibility and Incremental Deployment
+# Incremental Deployment {#rollout}
 
 Recall that TLS 1.2 permits computing the TLS PRF with any hash algorithm and PSK.
-Thus, an external PSK may be used with the same KDF (and underlying
-HMAC hash algorithm) as TLS 1.3 with importers. However, critically, the derived PSK will not
-be the same since the importer differentiates the PSK via the identity and hash function. Thus,
+Thus, an EPSK may be used with the same KDF (and underlying HMAC hash algorithm)
+as TLS 1.3 with importers. However, critically, the derived PSK will not be the same since
+the importer differentiates the PSK via the identity, target protocol, and target KDF. Thus,
```

Comment:
Does this text need to be tweaked if we believe EPSKs can't be used with TLS 1.2 at all?

Reply:
Hmm, I don't think so? What did you have in mind?